### PR TITLE
i18n(fr): Update `guides/routing.mdx` from #8941 and add `reference/api-reference.mdx` 

### PR DESCRIPTION
**Pull request by thomasbnt**
#### Description (required)
Update `guides/routing.mdx` from #8941

The link at 429 is good, see https://deploy-preview-8922--astro-docs-2.netlify.app/fr/reference/api-reference/#le-prop-page-de-pagination from https://github.com/withastro/docs/pull/8922

Add `reference/api-reference.mdx` with @ArmandPhilippot for the second part. 

#### Related issues & labels (optional)

- Suggested label: i18n

### CODE REVIEW

File: src/content/docs/fr/guides/routing.mdx
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/routing.mdx:429]: I haven't noticed it in `api-reference.mdx` but "*le prop*" sounds weird. Why not "*la prop*" instead of "*la propriété*" if we want to keep the abbreviation but this should be **la** and not **le** (in this PR and in #8922).

So, it should be:
```suggestion
<ReadMore>En savoir plus sur [la prop `page` de pagination](/fr/reference/api-reference/#la-prop-page-de-pagination).</ReadMore>
```

However, this PR might be merged before `api-reference.mdx` since this is a small one and #8922 is probably waiting for another reviewer. So I'm not sure we should use the translated anchor here. What do you think?

Comment (thomasbnt) [reply src/content/docs/fr/guides/routing.mdx:429]: > I haven't noticed it in `api-reference.mdx` but "_le prop_" sounds weird. Why not "_la prop_" instead of "_la propriété_" if we want to keep the abbreviation but this should be **la** and not **le** (in this PR and in #8922).
> 
> So, it should be:
> 
> However, this PR might be merged before `api-reference.mdx` since this is a small one and #8922 is probably waiting for another reviewer. So I'm not sure we should use the translated anchor here. What do you think?

yes, when I written this sentence, seems a little weird with `props`

Comment (thomasbnt) [reply src/content/docs/fr/guides/routing.mdx]: > However, this PR might be merged before api-reference.mdx since this is a small one and https://github.com/withastro/docs/pull/8922 is probably waiting for another reviewer. So I'm not sure we should use the translated anchor here. What do you think?
We can merge this one before #8922, but It won't interfere with the rest. It'll save you a second PR to change the link.

File: src/content/docs/fr/guides/routing.mdx
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/routing.mdx:429]: Since you choose `la propriété` instead of `la prop` in #8922 (see https://deploy-preview-8922--astro-docs-2.netlify.app/fr/reference/api-reference/#la-propri%C3%A9t%C3%A9-page-de-pagination) we need this update:
```suggestion
<ReadMore>En savoir plus sur [la prop `page` de pagination](/fr/reference/api-reference/#la-propriété-page-de-pagination).</ReadMore>
```

**Review by ArmandPhilippot — CHANGES_REQUESTED** [review]
I'm ok to leave the translated anchor, however it needs an update because of the change in #8922

**Comment by thomasbnt** [comment]
> I'm ok to leave the translated anchor, however it needs an update because of the change in #8922

yes, good!

**Review by ArmandPhilippot — APPROVED** [review]
LGTM, thanks @thomasbnt!

> [!note]
> The broken link will no longer be broken when #8922 is merged.

**Review by dreyfus92 — APPROVED** [review]
Thanks everybody! LGTM ✅

**Review by yanthomasdev — COMMENTED** [review]
LGTM! Could you actually move the changes here to #8922, so we can merge without link issues? It's a bit confusing since all the PRs open depend on one another, but I think getting these two together would unblock that.

**Review by yanthomasdev — COMMENTED** [review]
LGTM! Could you actually move the changes here to #8922, so we can merge without link issues? It's a bit confusing since all the PRs open depend on one another, but I think getting these two together would unblock that.

**Comment by thomasbnt** [comment]
> LGTM! Could you actually move the changes here to #8922, so we can merge without link issues? It's a bit confusing since all the PRs open depend on one another, but I think getting these two together would unblock that.

[Merged this PR](https://github.com/withastro/docs/pull/8922) into this one.

**Review by yanthomasdev — APPROVED** [review]
LGTM! Good job peeps!

**Pull request merged**
